tests/ResultSetTest.php

```diff
@@ -40,40 +57,49 @@ public function testErrorsShouldAdd(): void
         $this->assertContains('error2', $errors);
     }
 
-    public function testGetErrors(): void
+    public function testGetErrorObjects(): void
     {
-        $resultSet = new ResultSet();
-        $resultSet->addResult('attribute1', $this->createErrorResult('error1'));
-        $resultSet->addResult('attribute1', $this->createErrorResult('error2'));
-        $resultSet->addResult('attribute2', new Result());
-
-        $this->assertSame(
-            ['attribute1' => ['error1', 'error2']],
-            $resultSet->getErrors()
+        $this->assertEquals(
+            [
+                'attribute1' => [new Error('error1', []), new Error('error2', [])],
+                'attribute2' => [new Error('error3', ['path', 3])],
+            ],
+            $this->createErrorResultSet()->getErrorObjects()
         );
     }
 
-    public function testIsValidReturnTrue(): void
+    public function testGetErrors(): void
     {
-        $resultSet = new ResultSet();
-        $resultSet->addResult('attribute1', new Result());
-
-        $this->assertTrue($resultSet->isValid());
+        $this->assertEquals(
+            ['attribute1' => ['error1', 'error2'], 'attribute2' => ['error3']],
+            $this->createErrorResultSet()->getErrors()
+        );
     }
 
-    public function testIsValidReturnFalse(): void
+    public function testGetErrorsIndexedByPath(): void
     {
-        $resultSet = new ResultSet();
-        $resultSet->addResult('attribute1', $this->createErrorResult('error1'));
-
-        $this->assertFalse($resultSet->isValid());
+        $this->assertEquals(
+            ['attribute1' => ['' => ['error1', 'error2']], 'attribute2' => ['path.3' => ['error3']]],
```

Comment:
```suggestion
            ['attribute1' => ['attribute1' => ['error1', 'error2']], 'attribute2' => ['path.3' => ['error3']]],
```
`ResultSet` known about `attribute1` and must take this into account.

Reply:
So with this logic to be consistent for 'attribute2' it should be:

```php
'attribute2' => ['attribute.path.3' => ['error3']`
```

?

Reply:
Sorry. I'm was wrong.

Maybe this should be:

```php
[
    'attribute1' => ['error1', 'error2'],
    'attribute2.path.3' => ['error3'],
],
```

?

Reply:
:heavy_check_mark: